### PR TITLE
Add support for creating scheduled posts

### PR DESCRIPTION
**Pull request by BenLubar**

### CODE REVIEW

File: mastodon.go [hunk 1]
```diff
@@ -232,14 +232,23 @@ func (c *Client) authenticate(ctx context.Context, params url.Values) error {
...
-	Sensitive   bool   `json:"sensitive"`
-	SpoilerText string `json:"spoiler_text"`
-	Visibility  string `json:"visibility"`
+	Status      string    `json:"status"`
+	InReplyToID ID        `json:"in_reply_to_id"`
+	MediaIDs    []ID      `json:"media_ids"`
+	Sensitive   bool      `json:"sensitive"`
+	SpoilerText string    `json:"spoiler_text"`
+	Visibility  string    `json:"visibility"`
+	ScheduledAt time.Time `json:"scheduled_at,omitempty"`
```
Comment (mattn) [inline mastodon.go]: Should be a pointer?

Comment (BenLubar) [reply mastodon.go]: I don't think anyone will ever run into a situation where they want to schedule a post for midnight on January 1st 0001

Comment (mattn) [reply mastodon.go]: If post the Toot without initialize ScheduledA, how the mastodon server handle?

Comment (BenLubar) [reply mastodon.go]: since it has `,omitempty`, Mastodon will not receive a `scheduled_at` field if `ScheduledAt` is the zero value of `time.Time`.

Comment (mattn) [reply mastodon.go]: Really? Did you try it?

https://play.golang.org/p/cvY3Bx0-Cpd

Comment (BenLubar) [reply mastodon.go]: Oh wow, I've been wrong all this time about what omitempty actually does

(It seems to have a hard-coded list of things it considers to be "zero")

**Comment by mattn** [comment]
Thanks your fix. Please add test.

**Comment by mattn** [comment]
Thank you

**Pull request merged**